### ISSUE TITLE
`no-unsafe-property-key` cannot handle `unique symbol` correctly

### ISSUE THREAD

**Issue by E0SelmY4V** [sindresorhus/eslint-plugin-unicorn#3272]
The `no-unsafe-property-key` rule cannot correctly handle `unique symbol`.

`unique symbol` is not a template literal, does not have an `intrinsicName`, nor is it a default library symbol.
However, as a `symbol`, it has numerous properties, making it [isPossiblyUnsafePropertyKeyType](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/b8d96020a70efc90ed66e29b0d062561d083061a/rules/no-unsafe-property-key.js#L222-L223).

Nevertheless, it is actually safe.

It involves the `no-unsafe-property-key` rule.

```js
const uniqueSymbol = Symbol();
const myObject = { [uniqueSymbol]: true }; // that should be ok
```
